### DISCUSSION TITLE
Debian Update Steps

### DISCUSSION THREAD

**Discussion by nowjon** [teslamate-org/teslamate#1900] [Q&A]
Could we get steps added to the Debian install on how to upgrade? 
https://docs.teslamate.org/docs/installation/debian

Thanks!

**Comment by nowjon** [comment]
I found the update steps and tried to update following https://docs.teslamate.org/docs/upgrading/

I ran into these below issues, any advice? 

Database Migration Command
`root@Demeter:/usr/src/teslamate#  _build/prod/rel/teslamate/bin/teslamate eval "TeslaMate.Release.migrate"
ERROR! Config provider Config.Reader failed with:
** (ArgumentError) could not fetch environment variable "DATABASE_USER" because it is not set
    (elixir 1.12.0) lib/system.ex:699: System.fetch_env!/1
    (stdlib 3.15.1) erl_eval.erl:685: :erl_eval.do_apply/6
    (stdlib 3.15.1) erl_eval.erl:893: :erl_eval.expr_list/6
    (stdlib 3.15.1) erl_eval.erl:237: :erl_eval.expr/5
    (stdlib 3.15.1) erl_eval.erl:229: :erl_eval.expr/5
    (stdlib 3.15.1) erl_eval.erl:893: :erl_eval.expr_list/6
    (stdlib 3.15.1) erl_eval.erl:408: :erl_eval.expr/5
    (elixir 1.12.0) src/elixir.erl:280: :elixir.recur_eval/3

{"init terminating in do_boot",{#{'__exception__'=>true,'__struct__'=>'Elixir.ArgumentError',message=><<"could not                                                                                             fetch environment variable \"DATABASE_USER\" because it is not set">>},[{'Elixir.System','fetch_env!',1,[{file,"lib                                                                                            /system.ex"},{line,699}]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,685}]},{erl_eval,expr_list,6,[{file,"er                                                                                            l_eval.erl"},{line,893}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},{line,237}]},{erl_eval,expr,5,[{file,"erl_eval.er                                                                                            l"},{line,229}]},{erl_eval,expr_list,6,[{file,"erl_eval.erl"},{line,893}]},{erl_eval,expr,5,[{file,"erl_eval.erl"},                                                                                            {line,408}]},{elixir,recur_eval,3,[{file,"src/elixir.erl"},{line,280}]}]}}
init terminating in do_boot ({,[{Elixir.System,fetch_env!,1,[{_},{_}]},{erl_eval,do_apply,6,[{_},{_}]},{erl_eval,ex                                                                                            pr_list,6,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl_eval,expr,5,[{_},{_}]},{erl_eva

Crash dump is being written to: erl_crash.dump...done
root@Demeter:/usr/src/teslamate# LOGIN="admin:12131417" ./grafana/dashboards.sh restore

URL:                  http://localhost:3000
LOGIN:                admin:12131417
DASHBOARDS_DIRECTORY: ./grafana/dashboards

./grafana/dashboards.sh: line 112: jq: command not found
root@Demeter:/usr/src/teslamate# LOGIN="admin:admin" ./grafana/dashboards.sh restore

URL:                  http://localhost:3000
LOGIN:                admin:admin
DASHBOARDS_DIRECTORY: ./grafana/dashboards

./grafana/dashboards.sh: line 112: jq: command not found
`

TeslaMate Status
`demeter@Demeter:/usr/src/teslamate/_build/prod/rel/teslamate/bin$ sudo systemctl status teslamate
● teslamate.service - TeslaMate
     Loaded: loaded (/etc/systemd/system/teslamate.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2021-08-11 19:00:39 CDT; 59min ago
    Process: 1187 ExecStartPre=/usr/src/teslamate/_build/prod/rel/teslamate/bin/teslamate eval TeslaMate.Release.migrate (code=exited, status=0/SUCCESS)
   Main PID: 2261 (beam.smp)
      Tasks: 32 (limit: 50500)
     Memory: 190.6M
     CGroup: /system.slice/teslamate.service
             ├─2261 /usr/src/teslamate/_build/prod/rel/teslamate/erts-12.0.2/bin/beam.smp -- -root /usr/src/teslamate/_build/prod/rel/teslamate -progname erl -- -home /usr/src/teslamate -- -noshell -s elixi>
             ├─2339 /usr/src/teslamate/_build/prod/rel/teslamate/erts-12.0.2/bin/epmd -daemon
             ├─2360 erl_child_setup 1024
             ├─4011 inet_gethost 4
             └─4012 inet_gethost 4

Aug 11 19:31:58 Demeter teslamate[2261]: 2021-08-11 19:31:58.524 [error] ** (RuntimeError) Client Error: %{"error_type" => "client_error", "msg_type" => "data:error", "tag" => "200167189171", "value" => "ow>
Aug 11 19:31:58 Demeter teslamate[2261]:     (teslamate 1.23.7) lib/tesla_api/stream.ex:150: TeslaApi.Stream.handle_frame/2
Aug 11 19:31:58 Demeter teslamate[2261]:     (websockex 0.4.3) lib/websockex.ex:1113: WebSockex.try_callback/3
Aug 11 19:31:58 Demeter teslamate[2261]:     (websockex 0.4.3) lib/websockex.ex:797: WebSockex.common_handle/4
Aug 11 19:31:58 Demeter teslamate[2261]:     (stdlib 3.15.1) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
Aug 11 19:31:59 Demeter teslamate[2261]: 2021-08-11 19:31:59.002 [warn] GET https://owner-api.teslamotors.com/api/1/vehicles/212129141247/vehicle_data -> 503 (141.693 ms)
Aug 11 19:31:59 Demeter teslamate[2261]: 2021-08-11 19:31:59.002 [warn] TeslaApi.Error / "upstream connect error or disconnect/reset before headers. reset reason: connection failure"
Aug 11 19:31:59 Demeter teslamate[2261]: 2021-08-11 19:31:59.003 car_id=1 [error] Error / :unknown
Aug 11 19:32:39 Demeter teslamate[2261]: 2021-08-11 19:32:39.384 car_id=1 [info] Start / :online
Aug 11 19:32:39 Demeter teslamate[2261]: 2021-08-11 19:32:39.400 car_id=1 [info] Connecting ...
`

**Reply by EasternPA** [reply]
Is this related to the database being marked incompatible after the latest update? I got cocky and updated without backing up. And now the database is incompatible.
